packages/amplication-notifications-service/src/core/notifications/notifications.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Controller, Inject } from '@nestjs/common';
+import { MessagePattern, Payload } from '@nestjs/microservices';
+import {
+  INotification,
+  NOTIFICATION_TOKEN,
+} from 'src/contracts/interfaces/notification.interface';
+import { NotificationPattern } from 'src/contracts/notificationPattern';
+import { KafkaMessagePayload } from 'src/contracts/kafkaMessagePayload';
+import { EnvironmentVariables } from 'src/services/environmentVariables';
+
+const KAFKA_TOPIC_NAME_KEY = 'KAFKA_TOPIC';
+
+@Controller()
+export class NotificationsController {
+  constructor(
+    @Inject(NOTIFICATION_TOKEN) private notificationService: INotification
+  ) {}
+
+  @MessagePattern(EnvironmentVariables.getStrict(KAFKA_TOPIC_NAME_KEY))
+  async onNotificationReceived(
+    @Payload() message: { value: KafkaMessagePayload }
+  ) {
+    const { value } = message;
+    const notification: NotificationPattern = {
+      userId: value.userId,
+      notificationName: value.template,
+      payload: value.payload,
+    };
+    return this.notificationService.pushNotification(notification);
```

Comment:
We might want to implement a Dead Letter Queue here in case notifications fail. At the very least we should try-catch and log the error.

An alert of some sort would also be great, but maybe can be a part of a wider alerting strategy using Logz.io/CloudWatch Alerts/Discord or whatever other tools we have.

Reply:
@arielweinberger I wrapped it with try-catch and logged the error but I am not sure how to implement a Dead Letter Queue. I think that if this is something that we want we also need to implement it in the other services that consume Kafka queue

Reply:
I think we should be okay to push this one for now and deal with the DLQ later, just to unblock you. Next time we'll make sure it's accounted for in the design. Well done!